qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -391,3 +440,16 @@ def _compose_transforms(basis_transforms, source_basis, source_dag):
                 )
 
     return mapped_instrs
+
+
+def _compute_incomplete_basis(target):
+    incomplete_basis_gates = []
+    all_permutations_all_size = {}
+    for inst, qarg_props in target.items():
+        qargs = set(qarg_props)
+        size = len(next(iter(qargs)))
+        if size not in all_permutations_all_size:
+            all_permutations_all_size[size] = set(permutations(range(target.num_qubits), size))
+        if qargs != all_permutations_all_size[size]:
+            incomplete_basis_gates.append(inst)
```

Comment:
This is `O(n^d)` for a target with `n` qubits and a maximum gate arity of `d`.  I'm assuming that once we get to large backends, then they just won't be able to specify all properties for every pair (or even triple!) of qubits - even Eagle's going to be a bit of problem with ~16,000 elements necessary in these sets.  Are we allowed to assume that a target is well-formed?  If so, can we just skip creating our own permutation, and test the size of the set?

Also, are the qubits guaranteed to be ordered from 0 to `n - 1`?  I feel like I remember the qubit properties data structure being a dictionary, not an array.

Reply:
Yeah, I think I'm going to need to change to search for each edge (or multiqubit gate) in the target instead of trying to compute all the valid permutations and figure out if we need this or not. This is still WIP and something I realized a while ago (I should have left a comment) because also we're almost always going to be returning something for size >=2 here unless they have complete connectivity (which ions do at least). Since the full permutation map is all to all. I'm planning to update the logic here, I've just been concentrating on #5885 because that really needs to merge soon so we can unblock this and other things

Reply:
Sure, no problem - apologies if this review was a bit premature.

Reply:
Take a look at the current version (I force pushed it because I've been rebasing on #5885 as that changes) I think this should be more efficient than before. It still scales with the number of edges in the coupling map but not the full permutation. I did try adding a search for each edge and the performance was horrific on that.